src/auth/auth-store.tsx

```diff
@@ -32,11 +40,17 @@ interface AuthState {
 	updateEmail: (email: string) => Promise<void>;
 	updateUsername: (username: string) => Promise<void>;
 	deleteUser: () => Promise<void>;
+	adoptedTrees: Array<string> | null;
+	adoptedTreesInfo: Array<TreeInfo> | null;
```

Comment:
I think the typing  `| null` is not necessary in the two lines above. We can simply use an empty array as default, which makes the code also in the component much simpler.